cmd/main.go

```diff
@@ -2,39 +2,76 @@ package main
 
 import (
 	"context"
-	_ "embed"
 	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
 
-	"github.com/absmach/propeller/task"
-	"github.com/absmach/propeller/worker"
-	"github.com/google/uuid"
+	"github.com/absmach/propeller/proplet"
 )
 
-//go:embed add.wasm
-var addWasm []byte
-
 func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	t := task.Task{
-		ID:    uuid.New().String(),
-		Name:  "Addition",
-		State: task.Pending,
-		Function: task.Function{
-			File:   addWasm,
-			Name:   "add",
-			Inputs: []uint64{5, 10},
-		},
+	// Graceful shutdown on interrupt signal
+	go func() {
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+		<-sigChan
+		cancel()
+		fmt.Println("Shutting down gracefully...")
```

Comment:
You can use structured logging. You can have an example from https://github.com/absmach/propeller/blob/manager/cmd/manager/start.go#L45C1-L53C25